src/yarp.c

```diff
@@ -2180,8 +2181,8 @@ yp_parameters_node_location_set(yp_parameters_node_t *params, yp_node_t *param)
 // Append a required parameter to a ParametersNode node.
 static void
 yp_parameters_node_requireds_append(yp_parameters_node_t *params, yp_node_t *param) {
-  yp_parameters_node_location_set(params, param);
-  yp_node_list_append2(&params->requireds, param);
+  yp_parameters_node_location_set(params, (yp_node_t *) param);
```

Comment:
It looks like you shouldn't need to cast this since you already have a `yp_node_t *` right?